node/lib/util/write_repo_ast_util.js

```diff
@@ -399,6 +400,24 @@ git checkout-index -a -f
         }
     }
 
+    // set up a rebase if there is one
+
+    if (null !== ast.rebase) {
+        const rebase = ast.rebase;
+        const originalSha = commitMap[rebase.originalHead];
+        const ontoSha = commitMap[rebase.onto];
+        const original = yield NodeGit.AnnotatedCommit.lookup(repo,
+                                                              originalSha);
+        const onto = yield NodeGit.AnnotatedCommit.lookup(repo, ontoSha);
+        yield NodeGit.Rebase.init(repo, original, onto, null, null);
```

Comment:
Yeah, but here, you're passing "original" for branch and "onto" for "upstream" -- that seems wrong.